### ISSUE TITLE
`\mathpunct{}` and `\mathrel{}` and `\mathop{}` have zero width

### ISSUE THREAD

**Issue by hbghlyj** [ronkok/Temml#41]
<table>
<tr><td>In Temml, the three lines are the same</td><td>

![image](https://github.com/ronkok/Temml/assets/53823634/91253175-41b9-445f-a278-0930f9bcab29)

</td></tr>
<tr><td>In <a href=https://texlive.net/run?%5Cdocumentclass%7Barticle%7D%0A%5Cbegin%7Bdocument%7D%0A%24%5Cbegin%7Barray%7D%7Bl%7D%0Aab%0A%5C%5C%0Aa%5Cmathpunct%7B%7Db%0A%5C%5C%0Aa%5Cmathrel%7B%7Db%0A%5Cend%7Barray%7D%24%0A%5Cend%7Bdocument%7D>LaTeX</a>

```tex
\documentclass{article}
\begin{document}
$\begin{array}{l}
ab
\\
a\mathpunct{}b
\\
a\mathrel{}b
\end{array}$
\end{document}
```
</td><td>
The three lines are different:&nbsp;&nbsp;&nbsp; <img src=https://github.com/ronkok/Temml/assets/53823634/2d6fa8e5-639e-4d33-84cf-69334e03c4ad>
</td></tr>
</table>

Also, `\mathop{}` appear on page 5 of a [paper on TUGboat](http://tug.org/TUGboat/tb41-1/tb127gregorio-math.pdf)
> By the way, how to input the symbol in such
a way that the convention can be changed at will?
The simplest and more effective way is to define
`\newcommand{\diff}{\mathop{}\!d}`

but in Temml `\mathop{}` seems to have zero width.

**Comment by ronkok** [comment]
@hbghlyj, I have not been able to reproduce your first problem. This [demo page example](https://temml.org/#%5Cbegin%7Barray%7D%7Bl%7D%0Aab%0A%5C%5C%0Aa%5Cmathpunct%7B%7Db%0A%5C%5C%0Aa%5Cmathrel%7B%7Db%0A%5Cend%7Barray%7D) does have a variation in spacing, just like LaTeX.

With regard to your second problem, I agree that Temml is rendering `\mathop` with the wrong spacing. I'll look into it.

**Comment by hbghlyj** [comment]
> @hbghlyj, I have not been able to reproduce your first problem. This [demo page example](https://temml.org/#%5Cbegin%7Barray%7D%7Bl%7D%0Aab%0A%5C%5C%0Aa%5Cmathpunct%7B%7Db%0A%5C%5C%0Aa%5Cmathrel%7B%7Db%0A%5Cend%7Barray%7D) does have a variation in spacing, just like LaTeX.

Thanks, I find that it is a issue of Firefox. It has spacing in Chrome.

Firefox version 121.0

**Comment by ronkok** [comment]
Fixed the `\mathop{}` with release v0.10.20.
